pkg/common/morpc/client.go

```diff
@@ -235,7 +235,7 @@ func (c *client) getBackendLocked(backend string) (Backend, error) {
 		for i := uint64(0); i < n; i++ {
 			seq := c.mu.ops[backend].next()
 			b = backends[seq%n]
-			if b.LastActiveTime() != (time.Time{}) {
+			if !b.Locked() && b.LastActiveTime() != (time.Time{}) {
```

Comment:
I don't thik this code is correct.   Seems two clients, calling getBackendLocked, will get same b.

Later line 251 will crash.

Reply:
Never mind, you called c.mu.Lock before calling this, correct?   So far away ...

Reply:
no, two clients cannot call getBackendLocked at the same time. It's protected by s.mu.Lock